use-cases.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: page
+title: Instrumentation for Common Use Cases
+---
+<div id="toc"></div>
+
+This page aims to illustrate the common use cases that developers who instrument their applications with OpenTracing API need to deal with.
+
+### Tracing Server Endpoints
+
+When a server receives a request and needs to start tracing, it generally needs to go through three steps:
+
+  1. Attempt to deserialize Trace Context from the incoming request, in case the trace has already been started by the client.
+  1. Depending on the outcome, either start a new trace, or join the existing trace using deserialized Trace Context. This operation starts a new Span.
+  1. Store the newly created span in some _request context_ that is propagated throughout the application, either by application code, or by the RPC framework.
+
+#### Deserializing Trace Conext from Incoming Request
+
+Let's assume that we have an HTTP server, and the Trace Context is propagated from the client via HTTP headers, accessible via `request.headers`:
+
+```python
+    context = tracer.trace_context_from_text(
+        trace_context_id=request.headers,
+        trace_attributes=request.headers
+    )
+```
+
+Here we set both arguments of the decoding method to the `headers` map. The Tracer object will have the knowledge which headers it needs to read in order to reconstruct Trace Context, which is comprised of a trace context ID and trace attributes.
+
+#### Starting a Server-Side Span
+
+The `context` object above can be `None`, if the Tracer did not find relevant headers in the incoming request, either because the client did not send them, or maybe the client is running with a different tracer implementation (e.g. Zipkin vs. AppDash). In this case the server needs to start a brand new trace, otherwise it should join the existing trace:
+
+```python
+    if context is None:
+        span = tracer.start_trace(operation_name=operation)
+    else:
+        span = tracer.join_trace(operation_name=operation,
+                                 parent_trace_context=context)
+    span.set_tag('http.method', request.method)
+    span.set_tag('http.url', request.full_url)
+```
```

Comment:
Should we add a convenience `start_or_join_trace()` method that makes this top part a one-liner?